protocol.md

```diff
@@ -1323,6 +1324,45 @@ interface LogMessageParams {
 
 Where type is defined as above.
 
+#### <a name="window_progress"></a>Progress Notification
+
+The progress notification is sent from the server to the client to ask the client to indicate progress.
+
+_Notification_:
+* method: 'window/progress'
+* params: `ProgressParams` defined as follows:
+
+```typescript
+interface ProgressParams {
+  /**
+   * A unique identifier to associate multiple progress notifications with the same progress.
+   */
+  id: string;
+
+  /**
+   * The title of the progress.
+   * This should be the same for all ProgressParams with the same id.
+   */
+  title: string;
+
+  /**
+   * Optional progress message to display.
```

Comment:
We should think about this from a spec perspective and not necessarily about what VS Code does now.

Here is what I would propose:

1. Keep this optional.
2. If specified, it should override any previous progress message.
3. If not specified, the previous progress message is still valid.

If a server wants to clear the progress message without providing a new one, it can send the empty string.